extensions/AbuseFilter/AbuseFilter.hooks.php

```diff
@@ -100,7 +100,9 @@ public static function onAbortMove( $oldTitle, $newTitle, $user, &$error, $reaso
 		return $filter_result == '' || $filter_result === true;
 	}
 
-	public static function onArticleDelete( &$article, &$user, &$reason, &$error ) {
+	public static function onArticleDelete(
+		WikiPage $article, User $user, $reason, &$error
+	): bool {
```

Comment:
I'm not sure of the wider context here, but note that mandatory boolean returns not been present in MediaWiki for some time. This requirement was lifted in 2013 (release in MW 1.22) due to the common mistake where a hook is forgetting to return true/false (resulting in the implicit null return being casted to return false, which means to abort the action, and not allow any other hooks to run, either).

Given many hooks are just to allow modification to an array of some sorts (which cannot be aborted and has its return value ignored), this was removed to make the usage of those hooks more idiomatic.

This return type is probably safe to add here, but might conflict in the future or produce false negatives with regards to core not strictly expecting it (and not supposed to).